stl/inc/ranges

```diff
@@ -534,6 +535,332 @@ namespace ranges {
         inline constexpr _Single_fn single;
     } // namespace views
 
+    // CLASS TEMPLATE ranges::iota_view
+    template <class _Ty>
+    using _Iota_diff_t = conditional_t<is_integral_v<_Ty>, conditional_t<(sizeof(_Ty) < sizeof(int)), int, long long>,
+        iter_difference_t<_Ty>>;
+
+    // clang-format off
+    template <class _Ty>
+    concept _Decrementable = incrementable<_Ty> && requires(_Ty __t) {
+        { --__t } -> same_as<_Ty&>;
+        { __t-- } -> same_as<_Ty>;
+    };
+
+    template <class _Ty>
+    concept _Advanceable = _Decrementable<_Ty> && totally_ordered<_Ty> &&
+        requires(_Ty __i, const _Ty __j, const _Iota_diff_t<_Ty> __n) {
+            { __i += __n } -> same_as<_Ty&>;
+            { __i -= __n } -> same_as<_Ty&>;
+            _Ty(__j + __n);
+            _Ty(__n + __j);
+            _Ty(__j - __n);
+            { __j - __j } -> convertible_to<_Iota_diff_t<_Ty>>;
+        };
+
+    // clang-format off
+    template <weakly_incrementable _Wi>
+        requires semiregular<_Wi>
+    struct _Ioterator {
+        // clang-format on
+        /* [[no_unique_address]] */ _Wi _Current{};
+
+        using iterator_concept  = conditional_t<_Advanceable<_Wi>, random_access_iterator_tag,
+            conditional_t<_Decrementable<_Wi>, bidirectional_iterator_tag,
+                conditional_t<incrementable<_Wi>, forward_iterator_tag, input_iterator_tag>>>;
+        using iterator_category = input_iterator_tag;
+        using value_type        = _Wi;
+        using difference_type   = _Iota_diff_t<_Wi>;
+
+        _NODISCARD constexpr _Wi operator*() const noexcept(is_nothrow_copy_constructible_v<_Wi>) {
+            return _Current;
+        }
+
+        constexpr _Ioterator& operator++() noexcept(noexcept(++_Current)) /* strengthened */ {
+            ++_Current;
+            return *this;
+        }
+
+        constexpr auto operator++(int) noexcept(
+            is_nothrow_copy_constructible_v<_Wi>&& noexcept(++_Current)) /* strengthened */ {

```

Comment:
I've been making an effort to keep conditional `noexcept` simple, even to the extent of ignoring terms like `is_nothrow_copy_constructible_v` here which really only applies to one path of several. I'll tighten this up to `noexcept(++_Current) && (!incrementable<_Wi> || is_nothrow_copy_constructible_v<_Wi>)`, though since it bothers people.